File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/data/dto/discovery/ProductInfoDto.kt
package com.sudo248.sudoo.data.dto.discovery

import java.time.LocalDateTime

data class ProductInfoDto(
    val productId: String,
    val name: String,
    val sku: String,
    val images: List<String>,
    val price: Double,
    val listedPrice: Double,
    val amount: Int,
    val rate: Float,
    val discount: Int,
    val startDateDiscount: LocalDateTime,
    val endDateDiscount: LocalDateTime,
    val saleable: Boolean,
    val brand: String,
)

File: client/web/sudoo/lib/data/dto/discovery/product_info_dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:sudoo/domain/model/discovery/product_info.dart';

part 'product_info_dto.g.dart';

@JsonSerializable(explicitToJson: true)
class ProductInfoDto {
  final String productId;
  final String sku;
  final String name;
  final double price;
  final double listedPrice;
  final int amount;
  final double rate;
  final int discount;
  final DateTime? startDateDiscount;
  final DateTime? endDateDiscount;
  final bool saleable;
  final String brand;
  final List<String> images;

  const ProductInfoDto(
    this.productId,
    this.sku,
    this.name,
    this.price,
    this.listedPrice,
    this.amount,
    this.rate,
    this.discount,
    this.startDateDiscount,
    this.endDateDiscount,
    this.saleable,
    this.brand,
    this.images,
  );

  factory ProductInfoDto.fromJson(Map<String, dynamic> json) =>
      _$ProductInfoDtoFromJson(json);

  Map<String, dynamic> toJson() => _$ProductInfoDtoToJson(this);
}

extension Mapper on ProductInfoDto {
  ProductInfo toProductInfo() {
    return ProductInfo(
      productId,
      sku,
      name,
      price,
      listedPrice,
      amount,
      rate,
      discount,
      startDateDiscount,
      endDateDiscount,
      saleable,
      brand,
      images,
    );
  }
}
